examples/plugins/ebpf/http-requests.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
+import bcc
+
+import time
+import datetime
+import os
+import signal
+import errno
+import json
+import urlparse
+import threading
+import socket
+import BaseHTTPServer
+import SocketServer
+import sys
+
+EBPF_FILE = "http-requests.c"
+EBPF_TABLE_NAME = "received_http_requests"
+PLUGIN_UNIX_SOCK = "/var/run/scope/plugins/http_requests.sock"
+
+class KernelInspector(threading.Thread):
+    def __init__(self):
+        super(KernelInspector, self).__init__()
+        self.bpf = bcc.BPF(EBPF_FILE)
+        self.http_rate_per_pid = dict()
+        self.lock = threading.Lock()
+
+    def update_http_rate_per_pid(self, last_req_count_snapshot):
+        new_req_count_snapshot = dict()
+        new_http_rate_per_pid = dict()
+        req_count_table = self.bpf.get_table(EBPF_TABLE_NAME)
+        for key, value in req_count_table.iteritems():
+            request_delta = value.value
+            if key.pid in last_req_count_snapshot:
+                 request_delta -= last_req_count_snapshot[key.pid]
+            if request_delta > 0:
+                new_http_rate_per_pid[key.pid] = request_delta
+
+            new_req_count_snapshot[key.pid] = value.value
+
+        self.lock.acquire()
+        self.http_rate_per_pid = new_http_rate_per_pid
+        self.lock.release()
+
+        return new_req_count_snapshot
+
+    def on_http_rate_per_pid(self, f):
+        self.lock.acquire()
+        r = f(self.http_rate_per_pid)
+        self.lock.release()
+        return r
+
+    def run(self):
+        # Compute request rates based on the requests counts from the last
+        # second. It would be simpler to clear the table, wait one second
+        # clear() is expensive (each entry is individually cleared with a system
+        # call) and less robust (clearing contends with the increments done by
+        # the kernel probe).
+        # FIXME: we need a mechanism to garbage-collect old processes, either
+        #        here or on the probe. Some options are clearing the table once
+        #        in a while (not ideal for the reasons above) or adding another
+        #        probe to remove processes from the table when they die (this
+        #        will probably require keeping keeping track of tasks and not
+        #        just processes)
+        req_count_snapshot = dict()
+        while True:
+            time.sleep(1)
+            req_count_snapshot = self.update_http_rate_per_pid(req_count_snapshot)
+
+
+class PluginRequestHandler(BaseHTTPServer.BaseHTTPRequestHandler):
```

Comment:
My intention was to limit the dependencies as much as possible for now. At some point we may want to provide a library to simplify the plugins (then we can use flask or whathaveyou).

However, since we are using a container we could probably use whatever library we want already. I am not sure ...